src/solvers/prop/prop_conv.cpp

```diff
@@ -305,29 +305,13 @@ literalt prop_conv_solvert::convert_rest(const exprt &expr)
 
 bool prop_conv_solvert::set_equality_to_true(const equal_exprt &expr)
 {
-  if(!equality_propagation)
+  if(!equality_propagation || expr.lhs().id() != ID_symbol)
```

Comment:
Why is this changed -- what problem does this solve?

Reply:
Maybe it's subjective but I find the second version much more readable.
The problems were missing `const`, uninformative `result` variable name, confusing fall through in the case of `result.second` being false.